src/clapsolver/clap.py

```diff
@@ -22,14 +22,138 @@ def cost(i, j, costs):
         The total cost of the linear assignment problem solution under the
         constraint that row i is assigned to column j.
     """
-    costs = np.array(costs)
-    n_rows, n_cols = costs.shape
+    cost_matrix = np.array(cost_matrix)
+
+    if cost_matrix[i, j] == np.Inf or cost_matrix[i, j] == np.NAN:
+        return cost_matrix[i, j]
```

Comment:
How about this?
```python
if not np.isfinite(cost_matrix[i, j]):
    return cost_matrix[i, j]
```

Reply:
This doesn't quite cover all cases but it more concise than the code shown. An example case that is not handled correctly by this would be `i=0`, `j=0`, `cost_matrix=[[np.inf, 0],[0, -np.inf]]`, where the answer should be `np.nan`. However, I think it is okay to assume the matrix does not contain any `-np.inf` values and stick with the current implementation.